skore/src/skore/item/item_repository.py

```diff
@@ -64,33 +85,48 @@ def get_item(self, key) -> Item:
         Item
             The retrieved item.
         """
-        value = self.storage[key]
-        item_class_name = value["item_class_name"]
-        item_class = ItemRepository.ITEM_CLASS_NAME_TO_ITEM_CLASS[item_class_name]
-        item = value["item"]
+        return ItemRepository.__construct_item(self.storage[key][-1])
 
-        return item_class(**item)
+    def get_items(self, key) -> list[Item]:
+        """
+        Retrieve the list of items associated with `key` from the storage.
+
+        The list is ordered from oldest to newest "put" date.
+
+        Parameters
+        ----------
+        key : Any
+            The key used to identify the item in storage.
+
+        Returns
+        -------
+        list[Item]
+            The retrieved list of items.
+        """
+        return [ItemRepository.__construct_item(value) for value in self.storage[key]]
 
     def put_item(self, key, item: Item) -> None:
         """
         Store an item in storage.
 
+        This appends to a list of items previously associated with key `key`.
+
         Parameters
         ----------
         key : Any
             The key to use for storing the item.
         item : Item
             The item to be stored.
         """
-        item_parameters = item.__parameters__
+        item = ItemRepository.__deconstruct_item(item)
 
         if key in self.storage:
-            item_parameters["created_at"] = self.storage[key]["item"]["created_at"]
+            items = self.storage[key]
+            item["item"]["created_at"] = items[0]["item"]["created_at"]
```

Comment:
Last-minute thought: What about retro-compatibility in this special case?
Actually, if we run this code on an existing project instance, it will break because `self.storage[key]` is not a list.
The skore file must be removed.

Should-we leave a message or ensuring retro-compatibility with `if key in self.storage and type(self.storage[key]) == list:`?

Reply:
Good point; instinctively, I'm not sure we should add some code just for this purpose --- given the size of our user base, I think a message is more than enough.

On the other hand, I'm not sure how we could deal with this cleanly if we did have many users... some kind of DB migration?

Reply:
Status quo: we're still in the beta phase, which means we can introduce breaking-changes without backward compatibility.

@MarieS-WiMLDS, we should advise in the corresponding release of this feature that user must drop/recreate its project.

Reply:
Ok, I'll keep that in mind to say it in the release!  
Agree also that we can still do easily breaking changes for now.